File: sample/src/main/java/com/thoughtbot/expandablerecyclerview/sample/Genre.java
package com.thoughtbot.expandablerecyclerview.sample;

import com.thoughtbot.expandablerecyclerview.models.ExpandableGroup;
import java.util.List;

public class Genre extends ExpandableGroup<Artist> {

  private int iconResId;

  public Genre(String title, List<Artist> items, int iconResId) {
    super(title, items);
    this.iconResId = iconResId;
  }

  public int getIconResId() {
    return iconResId;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Genre)) return false;

    Genre genre = (Genre) o;

    return getIconResId() == genre.getIconResId();

  }

  @Override
  public int hashCode() {
    return getIconResId();
  }
}

File: sample/src/main/java/com/thoughtbot/expandablerecyclerview/sample/Genre.kt
package com.thoughtbot.expandablerecyclerview.sample

import com.thoughtbot.expandablerecyclerview.models.ExpandableGroup

class Genre(title: String?, items: List<Artist?>?, val iconResId: Int) : ExpandableGroup<Artist?>(title, items) {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is Genre) return false
        return iconResId == o.iconResId
    }

    override fun hashCode(): Int {
        return iconResId
    }
}